javascripts/govuk/show-hide-content.js

```diff
@@ -0,0 +1,150 @@
+(function (global) {
+  'use strict';
+
+  var $ = global.jQuery;
+  var GOVUK = global.GOVUK || {};
+
+  function ShowHideContent() {
+    var self = this;
+
+    // Radio and Checkbox selectors
+    var selectors = {
+      namespace: 'ShowHideContent',
+      radio: '.block-label input[type="radio"]',
+      checkbox: '.block-label input[type="checkbox"]'
+    };
+
+    // Escape name attribute for use in DOM selector
+    function escapeElementName(str) {
+      var result = str.replace('[', '\\[').replace(']', '\\]');
+      return(result);
+    }
+
+    // Adds ARIA attributes to control + associated content
+    function initToggledContent() {
+      var $control = $(this);
+      var $content = getToggledContent($control);
+
+      // Set aria-controls and defaults
+      if ($content.length) {
+        $control.attr('aria-controls', $content.attr('id'));
+        $control.attr('aria-expanded', 'false');
+        $content.attr('aria-hidden', 'true');
+      }
+    }
+
+    // Return toggled content for control
+    function getToggledContent($control) {
+      var $label = $control.parent('label');
+      return $('#' + $label.data('target'));
+    }
+
+    // Show toggled content for control
+    function showToggledContent($control, $content) {
+
+      // Show content
+      if ($content.hasClass('js-hidden')) {
+        $content.removeClass('js-hidden');
+        $content.attr('aria-hidden', 'false');
+
+        // If the controlling input, update aria-expanded
+        if ($control.attr('aria-controls')) {
+          $control.attr('aria-expanded', 'true');
+        }
+      }
+    }
+
+    // Hide toggled content for control
+    function hideToggledContent($control, $content) {
+      $content = $content || getToggledContent($control);
+
+      // Hide content
+      if (!$content.hasClass('js-hidden')) {
+        $content.addClass('js-hidden');
+        $content.attr('aria-hidden', 'true');
+
+        // If the controlling input, update aria-expanded
```

Comment:
I think I just found it odd reading that inputs without an 'aria-controls' attribute or a parent label with a `data-target` attribute were being sent to `getToggledContent` which assumes either of those.

At the moment this error is being absorbed by jQuery but it seems like this could be avoided by just filtering the radios down so only those controlling content had their content hidden.